service/storage_service.cc

```diff
@@ -564,7 +564,8 @@ future<storage_service::nodes_to_notify_after_sync> storage_service::sync_raft_t
             tmptr->add_leaving_endpoint(host_id);
             co_await update_topology_change_info(tmptr, ::format("{} {}/{}", rs.state, id, ip));
             break;
-        case node_state::replacing: {
+        case node_state::replacing:
+            if (rs.ring.has_value()) {
```

Comment:
Another reason why keeping this optional in the zero-token nodes PR is a good idea.